packages/db/tests/search/test-CollectionSearch.js

```diff
@@ -0,0 +1,308 @@
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 200,
+    updatedAt: 1579352701000,
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 50);
+});
+
+test('CollectionSearch supports term search for number field', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 200,
+    version: 2,
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 1);
+  t.is(response.results?.length, 1);
+});
+
+test('CollectionSearch supports term search for string field', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 200,
+    name: 'fakeCollection___0071',
+  };
+  const dbSearch2 = new CollectionSearch({ queryStringParameters });
+  const response2 = await dbSearch2.query(knex);
+  t.is(response2.meta.count, 1);
+  t.is(response2.results?.length, 1);
+
+  queryStringParameters = {
+    limit: 200,
+    process: 'publish',
+  };
+  const dbSearch3 = new CollectionSearch({ queryStringParameters });
+  const response3 = await dbSearch3.query(knex);
+  t.is(response3.meta.count, 50);
+  t.is(response3.results?.length, 50);
+});
+
+// TODO in CUMULUS-3639
+test.todo('CollectionSearch supports range search');
+
+test('CollectionSearch supports search for multiple fields', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 200,
+    name: 'testCollection___000',
+    updatedAt: 1579352700000,
+    process: 'ingest',
+    reportToEms: 'true',
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 1);
+  t.is(response.results?.length, 1);
+});
+
+test('CollectionSearch non-existing fields are ignored', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 200,
+    non_existing_field: `non_exist_${cryptoRandomString({ length: 5 })}`,
+    non_existing_field__from: `non_exist_${cryptoRandomString({ length: 5 })}`,
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 100);
+  t.is(response.results?.length, 100);
+});
+
+test('CollectionSearch returns fields specified', async (t) => {
+  const { knex } = t.context;
+  const fields = 'name,version,reportToEms,process';
+  const queryStringParameters = {
+    fields,
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 100);
+  t.is(response.results?.length, 10);
+  response.results.forEach((granule) => t.deepEqual(Object.keys(granule), fields.split(',')));
```

Comment:
should this be collection instead of granule? or am I misunderstanding what the results are here?